.github/workflows/ci-pr-validation.yaml

```diff
@@ -93,6 +93,14 @@ jobs:
       - name: Build
         run: make -j8
 
+      - name: Run OAuth2 tests
+        run: |
+          docker compose -f tests/oauth2/docker-compose.yml up -d
```

Comment:
I don't think so. Because the  set up phase has conflict with the default standalone (the JWT authn provider cannot support multiple public keys), so we have to set up another cluster or standalone for tests.

So we have to set up 3 standalones with different ports now. Ideally, we should not keep so many clusters running at the same time.

Currently, the `pulsar-tests` binary is already too huge. Each time a new file is added, the link time will increase. It's also better to split the whole binary into multiple binaries.

Reply:
Actually I'd like to separate the 2nd standalone into another `docker-compose.yml` like this PR. The 2nd standalone is added in https://github.com/apache/pulsar-client-cpp/pull/126, and it took much more time to run `pulsar-test-service-start.sh` so I pushed this PR for it: https://github.com/apache/pulsar-client-cpp/pull/236

In future, if we need to test standalone or cluster with different configs, setting up all these services in the same script is unnecessary. For example, we have no tests for geo-replication now. But setting up a Pulsar instance for multiple clusters is redundant for most of other tests.

Reply:
> So we have to set up 3 standalones with different ports now. Ideally, we should not keep so many clusters running at the same time.

Make sense to me. But, I'm worried that this is too unfriendly to other developers.

Maybe we should add scripts to run all unit tests in the future:

run_test.sh 
```
1. /pulsar-test-service-start.sh
2. Run common tests.
3. ./pulsar-test-service-stop.sh

4. start-mim-test-broker.sh
5. Run `HostNameValidation` relate test
6. stop-mim-test-broker.sh

7. start-oauth2-broker.sh
8. run oauth2 relate test
9. stop-mim-test-broker.sh

10. Run other special tests
```

I wouldn't block this PR

Reply:
> this is too unfriendly to other developers.

I agree. So I think we should add the documents for it. Actualy the scripts under the `build-support/` directory are already not friendly.

Reply:
> run_test.sh

I also thought of this idea just now. I will add the script in this PR.